### ISSUE TITLE
Shift-Tab completion doesn't work when edit cursor is at the start of the object

### ISSUE THREAD

**Issue by smcinerney** [jupyter/notebook#1902]
Shift-Tab completion on an object name only works when you place the edit cursor _inside_ or _after_ the object. Not when it's _at the start of the object_. I think that's contrary to most IDEs. Also, this gotcha is not documented anywhere.

```
import pandas
pd.crosstab
# ^ Put your edit cursor at the start of `pd.crosstab` and press Shift-Tab (on MacOS, Fn-Shift-Tab). Nothing happens. 
```

Version: Jupyter 4.2.3 on MacOS

**Comment by smcinerney** [comment]
Sorry, I'll refile this as a Jupyter issue.

**Comment by rgbkrk** [comment]
This is a good spot for the issue, assuming you're asking about the use inside the notebook.

**Comment by ellisonbg** [comment]
Good point, this might be in the kernel implementation of tab completion, but I suspect it is in the frontend.
